lib/serializer.js

```diff
@@ -43,3 +44,24 @@ Serializer.prototype.stringToByteArray = function (msg/* : string */) {
 Serializer.prototype.byteArrayToString = function (bytes/* : Uint8Array */) {
   return this.api.GenericMessage.decode(bytes).message
 }
+
+/**
+ * Serializes client sync credentials for accessing AWS and S3.
+ * @param {{aws: {{accessKeyId: <string>, secretAccessKey: <string>, sessionToken: <string>, expiration: <string>}}}, s3Post: {{bucket: <string>, postData: {{accessKeyId: <string>, secretAccessKey: <string>, sessionToken: <string>, expiration: <string>}}}}}} credentials
+ * @returns {Uint8Array}
+ */
+Serializer.prototype.credentialsToByteArray = function (credentials) {
+  return this.api.Credentials.encode({
+    aws: this.api.Credentials.lookup('Aws').create(credentials.aws),
+    s3Post: this.api.Credentials.lookup('S3Post').create(credentials.s3Post)
```

Comment:
so the fields are the actual types instead of Uint8Arrays with this the way it is now? would be good to have a test to prove that